pulsar-client-cpp/lib/ClientConnection.cc

```diff
@@ -114,13 +114,16 @@ isTlsAllowInsecureConnection_(false) {
     if (clientConfiguration.isUseTls()) {
         using namespace boost::filesystem;
 
-        boost::asio::ssl::context ctx(boost::asio::ssl::context::tlsv12_client);
-
+#if BOOST_VERSION >= 105400
+        boost::asio::ssl::context ctx(executor_->io_service_, boost::asio::ssl::context::tlsv12_client);
```

Comment:
Boost 1.41 doesn't has only one constructor for creating the context and this constructor io_service_

Current version of boost have rewritten the entire ssl code and don't require the io_service as a part of the constructor.